e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,373 @@
+	// Readiness behavior is undefined for this race condition.
+	// One or both of the Deployment and ReplicaSet may become unhealthy.
+	// But regardless, the user should be able to correct the situation.
+	nt.T.Log("Add a replicaset")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/replicaset-helloworld.yaml", yamlDir), "acme/namespaces/dir/replicaset.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add replicaset")
+
+	// This sync may block until reconcile timeout is reached,
+	// because ReplicaSet or Deployment may never reconcile.
+	// So this wait timeout must be longer than the reconcile timeout (5m).
+	nt.WaitForRepoSyncs()
+	nt.T.Log("check that the replicaset was created")
+	_, err := nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("get", "replicaset", "-n", "dir", "-l", "app=hello-world")
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Add a corresponding deployment")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add corresponding deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("check that the deployment was created")
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("get", "deployment", "hello-world", "-n", "dir")
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Remove the deployment")
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("check that the deployment was removed and replicaset remains")
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("get", "deployment", "hello-world", "-n", "dir")
+		return err
+	})
+	if err == nil {
+		nt.T.Fatal(err)
+	}
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("get", "replicaset", "-n", "dir", "-l", "app=hello-world")
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestRolebindingsUpdated(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.SkipMultiRepo)
+
+	nt.RootRepos[configsync.RootSyncName].Copy("../../examples/acme/namespaces/eng/backend/namespace.yaml", "acme/namespaces/eng/backend/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy("../../examples/acme/namespaces/eng/backend/bob-rolebinding.yaml", "acme/namespaces/eng/backend/br.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add bob rolebinding")
+	nt.WaitForRepoSyncs()
+	_, err := nt.Kubectl("get", "rolebinding", "-n", "backend", "bob-rolebinding")
+	if err != nil {
+		nt.T.Fatal("bob-rolebinding not found")
+	}
+
```

Comment:
Looks like this omits the `assert::contains "acme-admin"` assertion. Was that intentional?

Reply:
added predicate to test rolebinding name